tools/imglab/src/metadata_editor.cpp

```diff
@@ -656,7 +656,7 @@ on_overlay_rect_selected(
     const image_display::overlay_rect& orect
 )
 {
-    overlay_label.set_text(orect.label);
+    overlay_label.set_text(convert_utf8_to_utf32(orect.label));
```

Comment:
This still needed?

Reply:
Apparently, maybe I can call `convert_utf8_to_utf32` in this method, too.

Reply:
Oh right, `set_text()` for `std::string` calls `convert_mbstring_to_wstring()`.  I didn't write that unicode stuff.  When dlib switched away from subversion in 2008 all the version control history was lost :|

Anyway, yeah, I'm not super familiar with all the varied ways characters can be represented.  My impression is that everyone uses unicode, but :shrug:  There are several different versions of unicode and there is this whole multibyte/wchar_t thing which isn't unicode.  It's just some vague seemingly not well defined type.

Anyway, yeah `convert_utf8_to_utf32()` here seems very reasonable. I would say make that be what's inside `set_text()` but I don't know if that's going to break some existing user.

Reply:
Actually, I wanted to suggest dropping that wchar_t stuff and just support Unicode, haha... But yeah, it might break some people's workflow

Reply:
Yeah I am super confused about who uses `wchar_t`.  I think it's used a lot.  It's the standard type for this kind of thing.  But it's also under-specified, forcing people to do other stuff too.  I definitely don't want to drop it.  I bet people are using it.

Reply:
Ok, I will figure out a way to make it work when I get some time :)

Reply:
I was trying to figure out if Keita Mochizuki (the person who contributed all the nice unicode stuff) is on github or even what his email address is.  I apparently don't have that information anymore and can't find him :(

It was way back on sourceforge and there isn't any useful information there.  Back in the dark ages.

Reply:
Oh, that's too bad :(